Source/EphysSocket.cpp

```diff
@@ -162,51 +175,254 @@ bool EphysSocket::startAcquisition()
+    rc = socket->read(read_buffer.data(), HEADER_SIZE, true);
+
+    while (!stop_flag)
+    {
+        rc = socket->read(read_buffer.data(), matrix_size + HEADER_SIZE * num_expected_packets, true);
+
+        if (rc == -1)
+        {
+            if (socket->getRawSocketHandle() == -1)
+            {
+                CoreServices::sendStatusMessage("Ephys Socket: Socket handle is no longer valid.");
+                return;
+            }
+
+            CoreServices::sendStatusMessage("Ephys Socket: Data shape mismatch");
+            error_flag = true;
+            return;
+        }
+
+        for (int i = 0; i < num_expected_packets; i++)
+        {
+            header = Header(read_buffer, i * (max_data_size + HEADER_SIZE));
+
+            if (!compareHeaders(header))
+            {
+                CoreServices::sendStatusMessage("Ephys Socket: Header mismatch");
+                error_flag = true;
+                return;
+            }
+
+            if (i == 0 && header.offset != 0)
+            {
+                CoreServices::sendStatusMessage("Ephys Socket: Packets were dropped");
+                error_flag = true;
+                return;
+            }
+
+            int current_offset = HEADER_SIZE * (i + 1) + header.offset;
+
+            if (!buffer_flag) {
+                std::copy(read_buffer.begin() + current_offset, read_buffer.begin() + current_offset + header.num_bytes, recvbuf0.begin() + header.offset);
+            }
+            else {
+                std::copy(read_buffer.begin() + current_offset, read_buffer.begin() + current_offset + header.num_bytes, recvbuf1.begin() + header.offset);
+            }
+        }
+        
+        if (depth == U8) {
+            convertData<uint8_t>();
+        }
+        else if (depth == S8) {
+            convertData<int8_t>();
+        }
+        else if (depth == U16) {
+            convertData<uint16_t>();
+        }
+        else if (depth == S16) {
+            convertData<int16_t>();
+        }
+        else if (depth == S32) {
+            convertData<int32_t>();
+        }
+        else if (depth == F32) {
+            convertData<float_t>();
+        }
+        else if (depth == F64) {
+            convertData<double_t>();
+        }
+
+        full_flag = true;
+        buffer_flag = !buffer_flag;
+    }
+
+    return;
+}
+
+bool EphysSocket::updateBuffer()
+{
+    if (full_flag)
+    {
+        for (int i = 0; i < num_samp; i++) {
+            sampleNumbers.set(i, total_samples++);
+            ttlEventWords.set(i, eventState);
+        }
+
+        sourceBuffers[0]->addToBuffer(convbuf.data(),
+            sampleNumbers.getRawDataPointer(),
+            timestamps.getRawDataPointer(),
+            ttlEventWords.getRawDataPointer(),
+            num_samp,
+            1);
+
+        full_flag = false;
+    }
+
+    if (error_flag)
+        return false;
+    
+    else
+        return true;
+}
+
+String EphysSocket::handleConfigMessage(String msg)
```

Comment:
Is this a remote procedure call handler? It looks like its trying to set offset and scale parameters which we removed in this refactor, right?